tests/test_kratos_io.py

```diff
@@ -0,0 +1,206 @@
+                                         stress_initialisation_type=stress_initialisation_type,
+                                         time_integration=time_integration,
+                                         is_stiffness_matrix_constant=True, are_mass_and_damping_constant=True,
+                                         convergence_criteria=convergence_criterion,
+                                         strategy_type=strategy_type, scheme=scheme_type,
+                                         linear_solver_settings=linear_solver_settings, rayleigh_k=0.001,
+                                         rayleigh_m=0.001)
+
+        # set up problem data
+        return Problem(problem_name="test", number_of_threads=2, settings=solver_settings)
+
+    def test_write_project_parameters_json(
+        self,
+        create_default_2d_model_and_mesh:Model,
+        create_default_outputs:List[Output],
+        create_default_solver_settings:Problem
+    ):
+        """
+        Test correct writing of the project parameters for the default output, model and settings.
+
+        Args:
+            - create_default_2d_model_and_mesh (:class:`stem.model.Model`): the default 2D model of a square \
+                soil layer and a line load.
+            - create_default_outputs (List[:class:`stem.output.Output`]): list of default output processes.
+            - create_default_solver_settings (:class:`stem.solver.Problem`): the Problem object containing the \
+                solver settings.
+        """
+        model = create_default_2d_model_and_mesh
+        kratos_io = KratosIO(ndim=model.ndim)
+        model.project_parameters = create_default_solver_settings
+
+        actual_dict = kratos_io.write_project_parameters_json(
+            model=model,
+            outputs=create_default_outputs,
+            mesh_file_name="test_mdpa_file.mdpa",
+            materials_file_name="MaterialParameters.json",
+            output_folder="dir_test"
+        )
+        expected_dict = json.load(open("tests/test_data/expected_ProjectParameters.json", 'r'))
+        TestUtils.assert_dictionary_almost_equal(expected_dict, actual_dict)
+
+    def test_write_material_parameters_json(
+        self,
+        create_default_2d_model_and_mesh:Model
+    ):
+        """
+        Test correct writing of the material parameters for the default model.
+
+        Args:
+            - create_default_2d_model_and_mesh (:class:`stem.model.Model`): the default 2D model of a square \
+                soil layer and a line load.
+        """
+        model = create_default_2d_model_and_mesh
+        kratos_io = KratosIO(ndim=model.ndim)
+
+        actual_dict = kratos_io.write_material_parameters_json(model=model, output_folder="dir_test")
+        expected_dict = json.load(open("tests/test_data/expected_MaterialParameters.json", 'r'))
+        TestUtils.assert_dictionary_almost_equal(expected_dict, actual_dict)
+
+    def test_write_mdpa_file(
```

Comment:
can you also add a test for 3D

Reply:
only for the mpda, becase the material parameters and project parameters are already tested in other io's I think